### PR TITLE
[SPARK-27506][SQL][FOLLOWUP] Use option `avroSchema` to specify an evolved schema in `from_avro`

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

This is a follow-up of https://github.com/apache/spark/pull/26780
In https://github.com/apache/spark/pull/26780, a new Avro data source option `actualSchema` is introduced for setting the original Avro schema in function `from_avro`, while the expected schema is supposed to be set in the parameter `jsonFormatSchema` of `from_avro`. 

However, there is another Avro data source option `avroSchema`. It is used for setting the expected schema in readiong and writing.

This PR is to use the option `avroSchema` option for  reading Avro data with an evolved schema and remove the new one `actualSchema`

### Why are the changes needed?

Unify and simplify the Avro data source options.

### Does this PR introduce any user-facing change?

Yes.
To deserialize Avro data with an evolved schema, before changes:
```
from_avro('col, expectedSchema, ("actualSchema" -> actualSchema))
```

After changes:
```
from_avro('col, actualSchema, ("avroSchema" -> expectedSchema))
```

The second parameter is always the actual Avro schema after changes.
### How was this patch tested?

Update the existing tests in https://github.com/apache/spark/pull/26780

### CODE REVIEW

**Comment by gengliangwang** [comment]
@giamo @Fokko @cloud-fan @HyukjinKwon

**Comment by cloud-fan** [comment]
+1 to have less options

File: external/avro/src/main/scala/org/apache/spark/sql/avro/AvroOptions.scala [hunk 1]
```diff
@@ -36,18 +36,14 @@ class AvroOptions(
   }
 
   /**
-   * Optional schema provided by an user in JSON format.
+   * Optional schema provided by an user in JSON format. This option can be set when
+   * 1. for reading Avro, the expected schema for deserialization is compatible with
```
Comment (HyukjinKwon) [inline external/avro/src/main/scala/org/apache/spark/sql/avro/AvroOptions.scala]: This wording sounds a bit unclear to me.

> When reading, the provided Avro schema should be compatible with the Avro schema from the actual data.
> When writing, the provided Avro schema should be compatible with the Avro schema converted from Spark schema.

Is it correct?

File: external/avro/src/main/scala/org/apache/spark/sql/avro/AvroOptions.scala [hunk 2]
```diff
@@ -36,18 +36,14 @@ class AvroOptions(
   }
 
   /**
-   * Optional schema provided by an user in JSON format.
+   * Optional schema provided by an user in JSON format. This option can be set when
```
Comment (HyukjinKwon) [inline external/avro/src/main/scala/org/apache/spark/sql/avro/AvroOptions.scala]: nit: `an user` -> `a user`

**Review by HyukjinKwon — APPROVED** [review]
Looks good otherwise.

File: external/avro/src/main/scala/org/apache/spark/sql/avro/AvroOptions.scala (hunk 1 above)
Comment (gengliangwang) [reply external/avro/src/main/scala/org/apache/spark/sql/avro/AvroOptions.scala]: Yes. I was trying to explain the actual usage.
The doc is revised in my latest commit.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**